packages/cache-inmemory/src/services/inmemory-cache.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  ConfigurableModuleDeclaration,
+  MODULE_RESOURCE_TYPE,
+  TransactionBaseService,
+  ICacheService,
+} from "@medusajs/medusa"
+import { MedusaError } from "medusa-core-utils"
+import { EntityManager } from "typeorm"
+
+import { CacheRecord, InMemoryCacheModuleOptions } from "../types"
+
+const DEFAULT_TTL = 30 // seconds
+
+type InjectedDependencies = {}
+
+/**
+ * Class represents basic, in-memory, cache store.
+ */
+class InMemoryCacheService
+  extends TransactionBaseService
+  implements ICacheService
+{
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
+  protected TTL: number
+
+  protected readonly timoutRefs = {}
+  protected readonly store = new Map<string, CacheRecord<any>>()
+
+  constructor(
+    deps: InjectedDependencies,
+    options: InMemoryCacheModuleOptions = {},
+    moduleDeclaration?: ConfigurableModuleDeclaration
+  ) {
+    // @ts-ignore
+    super(...arguments)
+
+    this.TTL = options.defaultTTL || DEFAULT_TTL
+
+    if (moduleDeclaration?.resources !== MODULE_RESOURCE_TYPE.SHARED) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_ARGUMENT,
+        "At the moment this module can only be used with shared resources"
+      )
+    }
+  }
+
+  /**
+   * Retrieve data from the cache.
+   * @param key - cache key
+   */
+  async get<T>(key: string): Promise<T | null> {
+    const record: CacheRecord<T> | undefined = this.store.get(key)
+
+    if (!record || (record.expire && record.expire < Date.now())) {
```

Comment:
suggestion: store the now at the beginning of the method just for safety if it is close to the expire :D wdyt?